libs/pages/application/src/lib/feature/page-settings-healthchecks-feature/page-settings-healthchecks-feature.tsx

```diff
@@ -147,34 +117,54 @@ export function PageSettingsHealthchecksFeature() {
     methods.setValue('readiness_probe.current_type', defaultTypeReadiness)
     methods.setValue('liveness_probe.current_type', defaultTypeLiveness)
 
-    if (application?.healthchecks?.readiness_probe) {
-      setProbeValues('readiness_probe', application?.healthchecks?.readiness_probe)
+    if (service?.healthchecks?.readiness_probe) {
+      setProbeValues('readiness_probe', service?.healthchecks?.readiness_probe)
     }
 
-    if (application?.healthchecks?.liveness_probe) {
-      setProbeValues('liveness_probe', application?.healthchecks?.liveness_probe)
+    if (service?.healthchecks?.liveness_probe) {
+      setProbeValues('liveness_probe', service?.healthchecks?.liveness_probe)
     }
-  }, [methods, application, defaultTypeReadiness, defaultTypeLiveness])
+  }, [
+    methods,
+    service?.healthchecks?.liveness_probe,
+    service?.healthchecks?.readiness_probe,
+    defaultTypeReadiness,
+    defaultTypeLiveness,
+  ])
+
+  if (!service) return null
 
   return (
     <FormProvider {...methods}>
       <PageSettingsHealthchecks
-        ports={application?.ports}
-        jobPort={application?.port}
-        isJob={isJob(application)}
-        minRunningInstances={application?.min_running_instances}
+        ports={(service as Application | Container).ports}
+        jobPort={(service as Job).port}
+        isJob={service.serviceType === 'JOB'}
+        minRunningInstances={(service as Application | Container)?.min_running_instances}
         linkResourcesSetting={`${APPLICATION_URL(
           organizationId,
           projectId,
           environmentId,
           applicationId
         )}${APPLICATION_SETTINGS_URL}${APPLICATION_SETTINGS_RESOURCES_URL}`}
         onSubmit={onSubmit}
-        loading={loading}
+        loading={isLoadingEditService}
         environmentMode={environment?.mode}
       />
     </FormProvider>
   )
 }
 
-export default PageSettingsHealthchecksFeature
+export function PageSettingsAdvancedFeature() {
+  const { environmentId = '', applicationId = '' } = useParams()
+
+  const { data: serviceType } = useServiceType({ environmentId, serviceId: applicationId })
```

Comment:
Turn this hook into `useService`, then you can remove all  service null check in `SettingsHealthchecksFeature` and remove `useService` in it too

Reply:
https://github.com/Qovery/console/pull/1086/commits/2c3a612d072d328d714a86d7a67ea45427aed399